hazelcast/src/test/java/com/hazelcast/test/annotation/IsolatedJVMTest.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test.annotation;
+
+/**
+ * Annotates tests which has to be run in isolated (forked and fresh) JVM by surefire plugin in parallel profile.
+ * By default ParallelJVMTest are reusing forked JVMs and this can lead to problems with a static context.
+ * Combination of ParallelJVMTest (for example from parent class) and IsolatedJVMTest categories will prevent issue
+ * with a static context.
```

Comment:
What is the motivation for this category?

I understand it might be an easy way to achieve test isolation, but unfortunately, I think this is a dangerous tool to use. Forking the JVM for each such test is costly, for one test it will not matter, but such tests will inevitably accumulate over time and will worsen the situation with the slow PR builder and whole test suite.

What is the test we want to isolate in this case?

Reply:
These : https://github.com/hazelcast/hazelcast/pull/26032

Reply:
I wanted to add this due to test from this discussion:
https://github.com/hazelcast/hazelcast/pull/26032#discussion_r1401916639
but after comment from @k-jamroz :
`@Patras3 your change will fix the problem with tests, but we support embedded HZ instances. Nothing prevents user to start many embedded instances in the same JVM with different config. Or is such usage unsupported?`
I'm not sure now if this is a proper way. 
I wanted to use this annotation only for cases when tests fails in pr builder due to this issue.

Reply:
Ok, will comment on that PR.

Reply:
also some tests are heavy and misbehaving: #24563, isolating them would at least prevent cascading test failures (or maybe also decrease the likelihood of problems with the test itself) until the root cause is fixed

Reply:
>  slow PR builder

such tests will be the first candidate to optimize or remove from PR builder suite

Reply:
It's true that this can be used in way, also for investigation if test failure is due test itself or due to others, but then it cannot be used to "close" issue and if it will be a temporary solution then probably this annotation will remain forever 😅